libp2p/transports/tcptransport.nim

```diff
@@ -206,130 +208,121 @@ method accept*(self: TcpTransport): Future[Connection] =
   #      information is lost and must be logged here instead of being
   #      available to the caller - further refactoring should propagate errors
   #      to the caller instead
-  proc impl(
-      self: TcpTransport
-  ): Future[Connection] {.async: (raises: [transport.TransportError, CancelledError]).} =
-    proc cancelAcceptFuts() =
-      for fut in self.acceptFuts:
-        if not fut.completed():
-          fut.cancel()
-
-    if not self.running:
-      raise newTransportClosedError()
-
-    if self.servers.len == 0:
-      raise (ref TcpTransportError)(msg: "No listeners configured")
-    elif self.acceptFuts.len == 0:
-      # Holds futures representing ongoing accept calls on multiple servers.
-      self.acceptFuts = self.servers.mapIt(it.accept())
-
-    let
-      finished =
-        try:
-          # Waits for any one of these futures to complete, indicating that a new connection has been accepted on one of the servers.
-          await one(self.acceptFuts)
-        except ValueError:
-          raiseAssert "Accept futures should not be empty"
-        except CancelledError as exc:
-          cancelAcceptFuts()
-          raise exc
-      index = self.acceptFuts.find(finished)
-
-    # A new connection has been accepted. The corresponding server should immediately start accepting another connection.
-    # Thus we replace the completed future with a new one by calling accept on the same server again.
-    self.acceptFuts[index] = self.servers[index].accept()
-    let transp =
-      try:
-        await finished
-      except TransportTooManyError as exc:
-        debug "Too many files opened", description = exc.msg
-        return nil
-      except TransportAbortedError as exc:
-        debug "Connection aborted", description = exc.msg
-        return nil
-      except TransportUseClosedError as exc:
-        raise newTransportClosedError(exc)
-      except TransportOsError as exc:
-        raise (ref TcpTransportError)(msg: exc.msg, parent: exc)
-      except common.TransportError as exc: # Needed for chronos 4.0.0 support
-        raise (ref TcpTransportError)(msg: exc.msg, parent: exc)
-      except CancelledError as exc:
-        cancelAcceptFuts()
-        raise exc
 
-    if not self.running: # Stopped while waiting
-      await transp.closeWait()
-      raise newTransportClosedError()
+  proc cancelAcceptFuts() =
```

Comment:
I wonder what was the reason for having this logic in a proc within the `accept` method. The code is clearer now